Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -1274,10 +1185,120 @@ void function RemoveConnectToServerCallback( void functionref( ServerInfo ) call
 	file.connectCallbacks.fastremovebyvalue( callback )
 }
 
-void function TriggerConnectToServerCallbacks()
+void function TriggerConnectToServerCallbacks( ServerInfo ornull targetServer = null )
 {
+	ServerInfo server
+	if (targetServer == null)
+	{
+		targetServer = file.lastSelectedServer
+	}
+
 	foreach( callback in file.connectCallbacks )
 	{
-		callback( file.lastSelectedServer )
+		callback( expect ServerInfo( targetServer ) )
 	}
 }
+
+void function CancelJoinServer()
+{
+	file.cancelConnection = true
+}
+
+bool function JoinServer( ServerInfo server, string password = "" )
+{
+	if ( NSIsAuthenticatingWithServer() )
+		return false
+
+	print("Connecting to server " + server.name + " with id of " + server.id)
+	if ( password != "" )
+	{
+		print ( "password is: " + password )
+	}
+	TriggerConnectToServerCallbacks( server )
+	NSTryAuthWithServer( server.index, password )
+
+	ToggleConnectingHUD( true )
+
+	while ( NSIsAuthenticatingWithServer() && !file.cancelConnection )
+	{
+		WaitFrame()
+	}
+
+	ToggleConnectingHUD( false )
+
+	if ( file.cancelConnection )
+	{
+		file.cancelConnection = false
+		// re-focus server list
+		// Hud_SetFocused( Hud_GetChild( file.menu, "BtnServer" + ( file.serverButtonFocusedID + 1 ) ) )
```

Comment:
```suggestion
```
commented code that is no longer necessary?

Although I'm not sure if this code is actually unneeded, it may be worth moving it elsewhere